src/device/mod.rs

```diff
@@ -1,18 +1,40 @@
+use std;
+
 use manager;
 use manager::NetworkManager;
 
 
 #[derive(Debug)]
 pub struct Device {
-    interface: String,
-    path: String,
-    device_type: DeviceType,
-    state: DeviceState,
-    real: bool,
+    pub interface: String,
+    pub path: String,
+    pub device_type: DeviceType,
+    pub state: DeviceState,
+    pub real: bool,
 }
 
+impl Device {
+    pub fn from_path(manager: &NetworkManager, path: &String) -> Result<Device, String> {
```

Comment:
Should this be a dbus::Path instead?
https://docs.rs/dbus/0.5.2/dbus/struct.Path.html